### PR TITLE
Make sure the post content is loaded only after it's been saved to the DB

### LINKED ISSUE #4012: Reader: "Preview Post" (eye icon while you're editing a post) shows nothing the first time
### Expected behavior
 “preview post” (eye icon) should show the preview for the post you're editing.

### Actual behavior
it's showing nothing the first time, but works ok second time.

eye_icon_preview

### Steps to reproduce the behavior
1) write a post, enter a title and some content
2) tap on the preview (eye) icon to see the preview
3) the preview screen is blank

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #4012

To test: just start creating a new post and add a title, then click on the preview icon ("eye"). It should show the post contents (i.e., title if you only entered a title).

Needs review: @nbradbury

### CODE REVIEW

**Comment by nbradbury** [comment]
I'm not wild about changing `savePostAsync()` to do more than save the post. Having it update the preview fragment may fix this specific problem, but I think it complicates other uses.

For example, `savePostAsync()` is also called by `onSaveInstanceState()` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/4012-show-preview-post-first-time/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L389), and I don't think we want to update the preview fragment when `onSaveInstanceState()` is triggered.

**Comment by mzorz** [comment]
I've just made some changes @nbradbury let me know your thoughts!

**Comment by nbradbury** [comment]
> I've just made some changes @nbradbury let me know your thoughts!

Using a listener like that is exactly the change I had in mind, but note that [here](https://github.com/wordpress-mobile/WordPress-Android/pull/4017/commits/74fc31a8a1489fbc6d9b05233be95e14ad560d78#diff-38b6aea9cfff5f3e0e4b7389b44de602R822) the listener isn't being called on the main thread, which will likely cause problems.

**Comment by mzorz** [comment]
I'm having trouble seeing how that would be a problem/responsibility for the implementation of `savePostAsync` itself as pointed out in that link, so I still added a runOnUiThread() call but not on the `savePostAsync` implementation but only in the listener implementation instance itself (because the listener in savePostAsync doesn't need to impose any restrictions as to where the listener is ran, the implementation only needs that).

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -314,10 +314,21 @@ public void onPageSelected(int position) {
...
-                        mEditPostPreviewFragment.loadPost();
-                    }
+                    savePostAsync(new AfterSavePostListener() {
+                        @Override
+                        public void onPostSave() {
+                            if (mEditPostPreviewFragment != null) {
+                                runOnUiThread(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        if (mEditPostPreviewFragment != null) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: +1 on the additional null check now that we're on the UI thread.

**Comment by nbradbury** [comment]
My preference would be to have the listener fire on the UI thread rather than require the implementation instance to deal with the issue, but I realize I'm splitting hairs here so it's fine as-is.

I should mention, too, that this fix is only necessary with the "legacy" editor since the visual editor doesn't need a preview feature.

:shipit:

**Pull request merged**